examples/commandTest.lean

```diff
@@ -0,0 +1,95 @@
+import Qq.Commands
+import Qq.Delab
+
+open Qq Lean Elab Tactic
+
+macro "trace_state" : doElem => `(doElem| (by trace_state; exact pure ()))
+macro "trace_return" t:term : doElem => `(doElem| (by trace_state; exact pure $t))
+
+-- using the goal type (interpret 2, 3 as Int)
+/--
+trace: x : Q(Prop)
+inst✝ : Q(Decidable «$x»)
+⊢ TermElabM Q(Int)
+-/
+#guard_msgs in
+def f₁ (x : Prop) [Decidable x] : Int :=
+  by_elabq
+    trace_return q(if $x then 2 else 3)
+
+def add_self_37 {α : Type u} [Add α] (a : α) : α :=
+  by_elabq return (List.range 36).foldr (init := q($a)) fun _ acc => q($acc + $a)
```

Comment:
This test doesn't work if I replace `u` with `_`; do you agree that it should?

Reply:
In general, `by_elabq` cannot assign metavariables in the term. Is Qq even capable of that?

Reply:
for example, this fails too
```
def f₂ x := by_elabq return q(@id Nat x) -- resulting term contains metavariable
```
It fails already in the initial process of quoting the context. At some point, I thought I could fix that by setting anything with metavariables in their type as bare `Expr`, on the other hand, even then Qq doesn't unify on its own, e.g. this fails too
```
def f₂ x := by_elab -- failed to infer binder type
  let lctx ← getLCtx
  let x : Q(Nat) := Expr.fvar (lctx.findFromUserName? `x).get!.fvarId
  return q(@id Nat $x)
```

Reply:
Actually, it might not be that hard, just whenever unknown metavariables are encountered in a type / value of the decl, add them as free variables to the meta-context. Only in case there is a dependency between them, they must be added in the correct order. I can look at it at some point (but don't have time for it today).